### PR TITLE
chore(pkger): add stack state management for checks

### PR DESCRIPTION
**Pull request by jsteenb2**
references: #17434 

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass

### CODE REVIEW

File: cmd/influxd/launcher/pkger_test.go [hunk 1]
```diff
@@ -139,50 +129,64 @@ func TestLauncher_Pkger(t *testing.T) {
...
-				assert.Empty(t, sum.Buckets[0].Description)
+
+				require.Len(t, sum.Checks, 1)
+				assert.Equal(t, initialSum.Checks[0].Check.GetID(), sum.Checks[0].Check.GetID())
+				assert.Equal(t, updateCheckName, sum.Checks[0].Check.GetName())
 
 				require.Len(t, sum.Labels, 1)
 				assert.Equal(t, initialSum.Labels[0].ID, sum.Labels[0].ID)
 				assert.Equal(t, updateLabelName, sum.Labels[0].Name)
-				assert.Empty(t, sum.Labels[0].Properties.Color)
```
Comment (jsteenb2) [inline cmd/influxd/launcher/pkger_test.go:176]: not interested in checking these bits here, we have coverage below in spades

File: notification/check/check.go [hunk 2]
```diff
@@ -214,7 +214,8 @@ func UnmarshalJSON(b []byte) (influxdb.Check, error) {
 	}
 	if err := json.Unmarshal(b, &raw); err != nil {
 		return nil, &influxdb.Error{
-			Msg: "unable to detect the check type from json",
+			Code: influxdb.EInvalid,
```
Comment (jsteenb2) [inline notification/check/check.go:217]: useful messaging for downstream consumers, aka pkger looks wants to know that it failed b/c it is invalid, meaning, it is a check we can skip

File: pkger/parser.go [hunk 3]
```diff
@@ -594,9 +594,12 @@ func (p *Pkg) buckets(hideRemoved bool) []*bucket {
 	return buckets
 }
 
-func (p *Pkg) checks() []*check {
+func (p *Pkg) checks(hideRemoved bool) []*check {
```
Comment (jademcgough) [inline pkger/parser.go:597]: Nitpicky comment, but I typically prefer separate functions over boolean arguments (https://martinfowler.com/bliki/FlagArgument.html) to improve readability and detangle logic.

Comment (jsteenb2) [reply pkger/parser.go]: good call. Glad you pointed this out, definitely unneccessary. I just changed it to return all as it was initially doing. I do the filtering at callsite instead.

**Pull request merged**